src/group_theory/nilpotent.lean

```diff
@@ -738,3 +738,34 @@ begin
 end
 
 end classical
+
+lemma normalizer_condition_of_is_nilpotent [h : is_nilpotent G] : normalizer_condition G :=
+begin
+  -- roughly based on https://groupprops.subwiki.org/wiki/Nilpotent_implies_normalizer_condition
+  rw normalizer_condition_iff_only_full_group_self_normalizing,
+  unfreezingI
+  { induction h using nilpotent_center_quotient_ind with G' _ _ G' _ _ ih;
+    clear _inst_1 G; rename G' → G, },
+  { rintros H -, apply subsingleton.elim, },
+  { intros H hH,
+    by_cases hch : center G ≤ H,
```

Comment:
That's probably a `have`, not a by_cases, given the ex_falso in the second subgoals